src/FindARoommate/FindARoommate.js

```diff
@@ -24,27 +30,24 @@ export default function FindARoomate() {
   const classes = useStyles();
 
   const [semester, setSemester] = useState('');
-  const [residents, setResidents] = useState([]);
+  const [allResidents, setAllResidents] = useState([]);
   const [availableResidents, setAvailableResidents] = useState([]);
   const [selectedResident, setSelectedResident] = useState('');
+  const [profile, setProfile] = useState(false);
 
   useEffect(() => {
-    if (!availableResidents.length) {
-      getResidents();
-    } else {
-      getAvailableResidents(semester);
-    }
-  }, [])
+  
+  }, [semester, availableResidents])
```

Comment:
So clever! I'll have to put this one in my back pocket.

Reply:
The power of useEffect hook!

Reply:
Super powerful!